core/src/main/java/com/gentics/mesh/core/data/node/field/impl/NodeGraphFieldImpl.java

```diff
@@ -95,6 +98,17 @@
 			// Check whether the container already contains a node field
 			// TODO check node permissions
 			// TODO check whether we want to allow cross project node references
+
+			NodeFieldSchema nodeFieldSchema = (NodeFieldSchema) fieldSchema;
+			String schemaName = schema.getName();
```

Comment:
I think this is not correct. The name of the schema should be taken from the node.
E.g. node.getSchema().getName()

Reply:
HI @Jotschi , thanks for the comment. I did use String schemaName = node.getSchemaContainer().getName(); but with this, I always get the schema name as "folder" and my positive test cases fails because of this.
_org.junit.ComparisonFailure: 
Expected :The value for field "testField" must not use schema "test".
Actual   :The value for field "testField" must not use schema "folder"._

Reply:
Ok. Strange.I need to take a closer look and debug it inline. I'm however currently unable to do so. Maybe @philippguertler can give you feedback on this.

Reply:
@Jotschi is correct. You must use the schema of the referenced node to check here. It seems that the tests are wrong if this results in an error.